### PR TITLE
Vector Variable Support - Python

### PR DESCRIPTION
**Pull request by anujad95**
This has the required code changes for supporting python model based vector variables. It would still need some refinement and modifications.  This addresses #580.

### CODE REVIEW

**Comment by anujad95** [comment]
At this stage the PR is ready to be tested with python models with input vector variables for simulation, UQ, optimization, and surrogate modeling. Please test it and give feedback on improvements.

**Comment by anujad95** [comment]
At this stage the PR is ready to be tested with python models with input and output vector variables of any size, as per the users choice, for simulation, UQ, optimization, and surrogate modeling. Please test it and give feedback on improvements.

**Review by eslickj — COMMENTED** [review]
Looks good as far as  can tell.  Didn't test it, but I'm not likely to test it extensively enough to find a bug if there is one.  Let me know when you remove the extraneous code, and I'll do a more thorough review.

File: foqus_lib/framework/graph/graph.py [hunk 1]
```diff
@@ -401,9 +436,13 @@ def generateGlobalVariables(self):
         """
         self.x = self.input.createOldStyleDict()
         self.f = self.output.createOldStyleDict()
+        self.xvector = self.input_vectorlist.createOldStyleDict()
+        self.fvector = self.output_vectorlist.createOldStyleDict()
         # x and f are ordered dictionaries so keys are already sorted
         self.xnames = list(self.x.keys())  # get a list of input names
         self.fnames = list(self.f.keys())  # get a list of output names
+        # self.xvectornames = list(self.xvector.keys())  # get a list of input vector names
+        # self.fvectornames = list(self.fvector.keys())  # get a list of output vector names
```
Comment (eslickj) [inline foqus_lib/framework/graph/graph.py:445]: Go ahead and delete commented code.

**Review by eslickj — APPROVED** [review]
Looks good.

File: foqus_lib/framework/graph/graph.py (hunk 1 above)
Comment (anujad95) [reply foqus_lib/framework/graph/graph.py:445]: Yes I will delete this. Thank you @eslickj

**Review by Daison2102 — APPROVED** [review]
Tested two examples using the MATLAB-FOQUS interface and everything worked smoothly

**Pull request merged**